service/storage_proxy.cc

```diff
@@ -1909,14 +1909,18 @@ future<> storage_proxy::mutate_counters(Range&& mutations, db::consistency_level
         // so if there is a timeout we don't really know which one is to "blame"
         // and what to put in ks and cf fields of write timeout exception.
         // Let's just use the schema of the first mutation in a vector.
-        auto handle_error = [this, sp = this->shared_from_this(), s = endpoint_and_mutations.second[0].s, cl, permit] (std::exception_ptr exp) {
+        auto handle_error = [this, sp = this->shared_from_this(), s = endpoint_and_mutations.second[0].s, endpoint, cl, permit] (std::exception_ptr exp) {
             auto& ks = _db.local().find_keyspace(s->ks_name());
             try {
                 std::rethrow_exception(std::move(exp));
             } catch (rpc::timeout_error&) {
                 return make_exception_future<>(mutation_write_timeout_exception(s->ks_name(), s->cf_name(), cl, 0, db::block_for(ks, cl), db::write_type::COUNTER));
             } catch (timed_out_error&) {
                 return make_exception_future<>(mutation_write_timeout_exception(s->ks_name(), s->cf_name(), cl, 0, db::block_for(ks, cl), db::write_type::COUNTER));
+            } catch (rpc::closed_error&) {
+                const auto msg = fmt::format("Node {} closed connection while sending counter mutations", endpoint);
+                slogger.warn(msg.c_str());
```

Comment:
Why not do it in one line? Also connection closing should not cause WARN level messages.

Reply:
by the way, `slogger.warn` already accepts variadic arguments for formatting, no need to call fmt::format directly

Reply:
Or even any log. The node going down will be logged independently, and there can easily be tens of thousands of writes per second, or even more.

Reply:
I removed logger and inlined call to `format`.